File: src/CSP-AF/CSPAgents/CSPUtilityServerGroups/CSPClickAgent.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Models;
using CloudSuitePortal.Services;
using CloudSuitePortal.Controllers;

namespace actions.CSP.CSPAgents.CSPUtilityServerGroups
{
    class CSPClickAgent
    {
        public void run(Dictionary<string, object> Params)
        {
            var guid = (string)Params["GUID"];
            var index = Int32.Parse((string)Params["Index"]);
            var navigationController = CSPBrowser.Application.WindowController.NavigationController;
            var controller = (CSPUtilityServerGroupsController)navigationController.VisibleViewController;
            controller.ClickAgent(guid);
        }
    }
}

File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPClickAgent.java
package CSPAF.CSPAgents.CSPUtilityServerGroups;

import java.util.*;

import com.infor.cloudsuite.portal.core.services.*;
import com.infor.cloudsuite.portal.core.controllers.*;
import com.infor.cloudsuite.portal.singletenant.agents.controllers.*;

import org.openqa.selenium.NotFoundException;

class CSPClickAgent
{
    public void run(HashMap<String, Object> params) throws NotFoundException
    {
        String guid = params.get("GUID").toString();
        int index = Integer.parseInt(params.get("Index").toString());
        CSPNavigationController navigationController = CSPBrowser.getApplication().getWindowController().getNavigationController();
        CSPUtilityServerGroupsController controller = (CSPUtilityServerGroupsController)navigationController.getVisibleViewController();
        controller.clickAgent(guid);
    }
}